File: backend/internal/models/product.go
package models

import "time"

type Product struct {
	ID          uint             `gorm:"primaryKey" json:"id"`
	CategoryID  uint             `gorm:"not null" json:"category_id"`
	Barcode     string           `gorm:"unique;size:50" json:"barcode"`
	Name        string           `gorm:"not null;size:200" json:"name"`
	Price       int64            `gorm:"not null" json:"price"`
	CostPrice   int64            `gorm:"default:0" json:"cost_price"`
	Stock       int              `gorm:"not null;default:0" json:"stock"`
	MinStock    int              `gorm:"default:0" json:"min_stock"`
	Image       string           `gorm:"size:255" json:"image"`
	IsActive    bool             `gorm:"default:true" json:"is_active"`
	HasVariants bool             `gorm:"default:false" json:"has_variants"`
	CreatedAt   time.Time        `json:"created_at"`
	UpdatedAt   time.Time        `json:"updated_at"`
	Category    Category         `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Variants    []ProductVariant `gorm:"foreignKey:ProductID" json:"variants,omitempty"`
}

File: tailadmin/src/types/product.ts
export interface Category {
  id: number;
  name: string;
  description: string;
  created_at: string;
}

export interface ProductVariant {
  id: number;
  product_id: number;
  name: string;
  barcode: string;
  price: number;
  stock: number;
  min_stock: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface VariantInput {
  name: string;
  barcode?: string;
  price: number;
  stock?: number;
  min_stock?: number;
}

export interface Product {
  id: number;
  category_id: number;
  barcode: string;
  name: string;
  price: number;
  cost_price: number;
  stock: number;
  min_stock: number;
  image: string;
  is_active: boolean;
  has_variants: boolean;
  created_at: string;
  category?: Category;
  variants?: ProductVariant[];
}

export interface ProductInput {
  category_id: number;
  barcode?: string;
  name: string;
  price: number;
  cost_price?: number;
  stock?: number;
  min_stock?: number;
  image?: string;
  is_active?: boolean;
}
